### DISCUSSION TITLE
Query Table Relationships to Hide Table Row

### DISCUSSION THREAD

**Discussion by Al-Blu** [Budibase/budibase#5176] [Help]
Hey, 
I am build a Rating App for a company. I am on the last hurdle but am stuck stuck. I posted a question already(#5064) but think what I was asking was too complicated. So I will try again with images.

![BudiBaseQuestion](https://user-images.githubusercontent.com/99728863/160846002-96d14f9c-1c59-4327-b1d9-799b4354a686.png)

Any guidance would be much appreciated.
Thanks

**Comment by melohagan** [comment]
Hey @Al-Blu 

For more complicated relational functionality, I recommend using MySQL or PostgreSQL.

Here's an example in MySQL to accomplish what you are looking for:

**Step 1 - Schema**
 
Create your tables in MySQL:

```
CREATE TABLE `sys`.`UseCases` (
  `idUseCases` INT NOT NULL AUTO_INCREMENT,
  `Name` VARCHAR(45) NULL,
  `Description` VARCHAR(45) NULL,
  PRIMARY KEY (`idUseCases`));
```

```
CREATE TABLE `sys`.`SurveyResults` (
  `idSurveyResults` INT NOT NULL AUTO_INCREMENT,
  `idUseCases` INT NULL,
  `Sum` INT NULL,
  `Email` VARCHAR(45) NULL,
  `idUser` VARCHAR(145) NULL,
  PRIMARY KEY (`idSurveyResults`));
```

**Step 2 - Create your MySQL query**

Add a MySQL datasource and fetch your tables (docs <a href="https://docs.budibase.com/docs/postgresql">here</a>)

![Screenshot 2022-03-31 at 14 17 23](https://user-images.githubusercontent.com/101575380/161063791-159986be-8bc8-48ff-9a2a-d2032824c8ef.png)

Click 'Run Query', amend the 'Schema' and then 'Save Query'

The SQL Query returns use cases that have not received any survey responses for the given *USER_ID* binding:
```
SELECT idUseCases, Name, Description, COUNT(idUseCases) FROM (
SELECT DISTINCT UseCases.*,
CASE WHEN SurveyResults.idUser = '{{ USER_ID }}' THEN 1
END AS UseCaseUserMatch
FROM sys.UseCases
JOIN sys.SurveyResults on SurveyResults.idUseCases = UseCases.idUseCases) as alias
GROUP BY idUseCases
HAVING COUNT(idUseCases) < 2
```

**Step 3 - Design**

Add a Data Provider for your UseCases table, and select the query as the data source. Then click to cog icon to set the binding to the current user id:

![Screenshot 2022-03-31 at 14 21 15](https://user-images.githubusercontent.com/101575380/161064498-f4e27a60-9296-44d4-87fc-15721f7067ac.png)

Next link the Add Input button to your new survey result screen, but make sure to include the **idUseCases** in the path:

![Screenshot 2022-03-31 at 14 26 43](https://user-images.githubusercontent.com/101575380/161065571-34dcced7-98ba-4f0e-ac30-0fd6d9a8e94d.png)

The route in your New SurveyResults screen should appear as follows:
![Screenshot 2022-03-31 at 14 27 56](https://user-images.githubusercontent.com/101575380/161065831-080a21a8-30e3-4a26-8657-b62449872be2.png)

Finally, add the **idUseCases** and **idUser** fields into the 'Save Row' action of the Save button. You can delete these fields from the form view itself.

![Screenshot 2022-03-31 at 14 46 30](https://user-images.githubusercontent.com/101575380/161070333-33eecb61-aafc-47b8-aeca-c326a9cf40c3.png)

**Comment by Al-Blu** [comment]
Thanks for the reply. Totally appreciate it. I will get on it right away. There's a bit of a learn curve for me here. Cheers.
